activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -608,10 +608,10 @@ def primary_keys(table_name) # :nodoc:
       end
 
       def case_sensitive_comparison(table, attribute, column, value)
-        if value.nil? || column.case_sensitive?
-          super
-        else
+        if !value.nil? && column.collation && !column.case_sensitive?
```

Comment:
Current behavior is:

* In the case of a column has `collation`:
 * `case_sensitive?` returns `false` when the suffix of `collation` includes `_ci` (e.g. `utf8_general_ci`) (**A**)
 * `case_sensitive?` returns `true` when the suffix of `collation` doesn't include `_ci` (e.g. `utf8_bin`) (**B**)
* In the case of a column doesn't have `collation`:
 * `case_sensitive?` returns always `false` (**C**)

The BINARY Operator is only needed in the case of **A**, not needed for **B** and **C**.